dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.cpp

```diff
@@ -1324,6 +1322,53 @@ bool RtpsUdpDataLink::requires_inline_qos(const GUIDSeq_var& peers)
 
 bool RtpsUdpDataLink::force_inline_qos_ = false;
 
+void
+RtpsUdpDataLink::RtpsWriter::send_heartbeats(const DCPS::MonotonicTimePoint& now)
+{
+  OPENDDS_VECTOR(TransportQueueElement*) pendingCallbacks;
+
+  RtpsUdpDataLink_rch link = link_.lock();
+  if (!link) {
+    return;
+  }
+
+  RtpsUdpInst& cfg = link->config();
+
+  {
+    ACE_GUARD(ACE_Thread_Mutex, g2, heartbeat_mutex_);
+    if (expected_acks_) {
+      // One more readers did not ack since the last heartbeat.
+      heartbeat_period_ *= cfg.heartbeat_backoff_factor_;
+    } else if (last_ack_ != MonotonicTimePoint::zero_value && last_heartbeat_ != MonotonicTimePoint::zero_value) {
+      heartbeat_period_ = (last_ack_ - last_heartbeat_) * cfg.heartbeat_safety_factor_;
+    }
+    heartbeat_period_ = std::min(heartbeat_period_, std::max(cfg.heartbeat_period_maximum_, cfg.heartbeat_period_minimum_));
```

Comment:
It seems like the relative order of `cfg.heartbeat_period_maximum_, cfg.heartbeat_period_minimum_` should already be known

Reply:
There is nothing that prevents the user from misconfiguring it and I didn't see a fool-proof method for correcting it when they do.

Reply:
Can you fix the order in the transport constructor so you only have to do it once per run and not for every heartbeat?